component-interaction/src/app/user-list/user-list.component.html

```diff
@@ -1,5 +1,8 @@
 <h2>Users and their details:</h2>
 <button class="success" (click)="handleAddClick()">Add User</button>
-<user-details *ngFor="let user of allUsers" [user]="user" (onDeleteUser)="handleDeleteUser($event)">
+<button class="rm-selected" *ngIf="existsSelected()" (click)="deleteSelectedUsers()">Del

```

Comment:
*ngIf="existsSelected()"

for ngIfs try to have an instruction instead of a function. If you have more than an instruction, use pipes (I will shou you about pipes). You need this, because ngIf will call "existsSelected()" a lot

Reply:
add console.log() inside "existsSelected()" and check how many times it will be shown :))

Reply:
for your case, *ngIf="selectedUsersId.length > 0" is perfect